components/user/SignupForm.vue

```diff
@@ -0,0 +1,183 @@
+<script setup lang="ts">
+  import {
+    useRequiredInput,
+    useValidateEmail,
+    useValidatePassword,
+    useRequireEqualPasswords,
+  } from '@/composables/useForm'
+
+  const initialState = {
+    name: '',
+    email: '',
+    password: '',
+    passwordConfirmation: null,
+    studyProgram: '',
+    currentYear: '',
+    userType: null,
+    registrationCode: null,
+  }
+
+  const state = reactive({
+    ...initialState,
+  })
+
+  const form = ref()
+  const submit = async () => {
+    if (!form.value) return
+    const { valid } = await form.value.validate()
+
+    if (!valid) throw new Error('Form is not valid')
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const { passwordConfirmation, userType, ...user } = state
+
+    if (userType === 'student') {
+      await registerUser({

```

Comment:
The registerUser function will throw errors if something goes wrong. These must be handled with _trycatch_ blocks and proper feedback to the user through some sort of popup.